py/objtype.h

```diff
@@ -37,6 +37,11 @@ typedef struct _mp_obj_instance_t {
     // TODO maybe cache __getattr__ and __setattr__ for efficient lookup of them
 } mp_obj_instance_t;
 
+// Flags for type behaviour
+#define TYPE_FLAG_IS_SUBCLASSED (0x0001)
+#define TYPE_FLAG_HAS_SPECIAL_ACCESSORS (0x0002)
+#define TYPE_FLAG_NO_EQUALITY_SHORTCUTS (0x0004)
```

Comment:
I would say these should go in `py/obj.h` because that's where `mp_obj_type_t` is defined (yes that file is large and a bit messy but it contains the core definitions and now that these flags are needed outside `objtype.c` I think they should be available for everyone without needing to include `py/objtype.h`).

Also, they should probable be prefixed with `MP_` now that they will be in a public header.

Reply:
Makes sense. Done.